server/src/main/java/org/opensearch/gateway/remote/model/RemoteClusterMetadataManifest.java

```diff
@@ -0,0 +1,153 @@
+            DELIMITER,
+            MANIFEST_FILE_PREFIX,
+            RemoteStoreUtils.invertLong(clusterMetadataManifest.getClusterTerm()),
+            RemoteStoreUtils.invertLong(clusterMetadataManifest.getStateVersion()),
+            (clusterMetadataManifest.isCommitted() ? "C" : "P"), // C for committed and P for published
+            RemoteStoreUtils.invertLong(System.currentTimeMillis()),
+            String.valueOf(clusterMetadataManifest.getCodecVersion()) // Keep the codec version at last place only, during read we reads
+                                                                      // last place to
+            // determine codec version.
+        );
+        this.blobFileName = blobFileName;
+        return blobFileName;
+    }
+
+    @Override
+    public UploadedMetadata getUploadedMetadata() {
+        assert blobName != null;
+        return new UploadedMetadataAttribute(MANIFEST_PATH_TOKEN, blobName);
+    }
+
+    @Override
+    public InputStream serialize() throws IOException {
+        return CLUSTER_METADATA_MANIFEST_FORMAT.serialize(
+            clusterMetadataManifest,
+            generateBlobFileName(),
+            getCompressor(),
+            RemoteClusterStateUtils.FORMAT_PARAMS
+        ).streamInput();
+    }
+
+    @Override
+    public ClusterMetadataManifest deserialize(final InputStream inputStream) throws IOException {
+        ChecksumBlobStoreFormat<ClusterMetadataManifest> blobStoreFormat = getClusterMetadataManifestBlobStoreFormat();
+        return blobStoreFormat.deserialize(blobName, getNamedXContentRegistry(), Streams.readFully(inputStream));
+    }
+
+    private int getManifestCodecVersion() {
+        assert blobName != null;
+        String[] splitName = blobName.split(DELIMITER);
+        if (splitName.length == SPLITTED_MANIFEST_FILE_LENGTH) {
+            return Integer.parseInt(splitName[splitName.length - 1]); // Last value would be codec version.
+        } else if (splitName.length < SPLITTED_MANIFEST_FILE_LENGTH) { // Where codec is not part of file name, i.e. default codec version 0
+            // is used.
+            return ClusterMetadataManifest.CODEC_V0;
+        } else {
+            throw new IllegalArgumentException("Manifest file name is corrupted");
+        }
+    }
+
+    private ChecksumBlobStoreFormat<ClusterMetadataManifest> getClusterMetadataManifestBlobStoreFormat() {
+        long codecVersion = getManifestCodecVersion();
+        if (codecVersion == MANIFEST_CURRENT_CODEC_VERSION) {
+            return CLUSTER_METADATA_MANIFEST_FORMAT;
+        } else if (codecVersion == ClusterMetadataManifest.CODEC_V1) {
+            return CLUSTER_METADATA_MANIFEST_FORMAT_V1;
+        } else if (codecVersion == ClusterMetadataManifest.CODEC_V0) {
+            return CLUSTER_METADATA_MANIFEST_FORMAT_V0;
+        }
+        throw new IllegalArgumentException("Cluster metadata manifest file is corrupted, don't have valid codec version");
```

Comment:
Lets add the found version in the exception message

Reply:
Ack